modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/transaction/DistributedTransaction.java

```diff
@@ -60,18 +76,63 @@ public boolean isTransactional() {
         return false;
     }
 
-    @Override public void onClose() {
-
+    @Override
+    public void onClose() {
     }
 
     @Override
-    public void start(Xid xid, boolean join, boolean resume) {
-
+    public void start(Xid xid, int sessionId, boolean join, boolean resume) throws ValidationException {
+        if (join && resume) {
+            throw new ValidationException("Cannot start a branch with both join and resume set " + xid);
+        }
+
+        Branch branch = transactionRegistry.getBranch(xid);
```

Comment:
Cant we remove the varibale 'branch' and directly do 
`currentBranch = transactionRegistry.getBranch(xid);`
It can reduce several lines below as well?

Reply:
Purpose of current branch is to identify the enqueue dequeue strategy of the transaction. When the transaction is within a start and end block messages get setup for a transaction commit / rollback. If it is outside a transaction, messages get directly enqueued or dequeued. Therefore we set the enqueue dequeue strategy (current branch) only if we successfully start a transaction.

Reply:
+1